Extractor/Nodes/BaseUANode.cs

```diff
@@ -562,6 +563,84 @@ protected Dictionary<string, string> BuildMetadataBase(Dictionary<string, string
             return result;
         }
 
+        protected Dictionary<string, JsonNode?> BuildMetadataJsonBase(Dictionary<string, JsonNode?>? extras, IUAClientAccess client, bool reversibleJson = true)
+        {
+            var result = extras ?? new Dictionary<string, JsonNode?>();
+
+            if (Properties == null) return result;
+
+            var fields = new HashSet<string>(result.Keys);
+            foreach (var prop in Properties)
+            {
+                if (prop != null && !string.IsNullOrEmpty(prop.Name))
+                {
+                    var name = prop.Name;
+                    string safeName = name;
+                    int idx = 0;
+                    while (!fields.Add(safeName))
+                    {
+                        safeName = $"{name}{idx++}";
+                    }
+                    result[safeName] = CreatePropertyJsonValue(prop, client, reversibleJson);
+                }
+            }
+            return result;
+        }
+
+        private JsonNode? CreatePropertyJsonValue(BaseUANode prop, IUAClientAccess client, bool reversibleJson)
+        {
+            // Get nested properties
+            var nestedProperties = prop.Properties != null ?
+                prop.BuildMetadataJsonBase(null, client, reversibleJson) :
+                new Dictionary<string, JsonNode?>();
+
+            // Following UAPropertySerializer pattern: if no nested content, return value directly
+            if (nestedProperties.Count == 0)
+            {
+                if (prop is UAVariable variable)
+                {
+                    if (variable.Value == null)
+                        return null;
+
+                    return client.TypeConverter.ConvertToJson(
+                        variable.Value.Value,
+                        variable.FullAttributes.DataType?.EnumValues,
+                        mode: reversibleJson ? JsonMode.Json : JsonMode.ReversibleJson);
+
+                }
+                return null;
+            }
+
+            // Has nested properties - create object with Value field
+            var propertyObject = new JsonObject();
+
+            // Add main value
+            if (prop is UAVariable variable1)
+            {
+                if (variable1.Value == null)
+                    return null;
+
+                propertyObject["Value"] = client.TypeConverter.ConvertToJson(
+                    variable1.Value.Value,
+                    variable1.FullAttributes.DataType?.EnumValues,
+                    mode: reversibleJson ? JsonMode.Json : JsonMode.ReversibleJson);
+            }
+
+            // Add nested properties, deduplicating conflicts
+            foreach (var nestedProp in nestedProperties)
+            {
+                var name = nestedProp.Key;
+                string safeName = name;
+                int idx = 0;
+                while (propertyObject.ContainsKey(safeName))
```

Comment:
Will the duplicates appear often?

If so the while-loop would often run many iterations, and increase the idx only be 1 for every iteration.
Should it instead use a random value?

Reply:
Looks like you didn't go for a random value, but just to be clear, the consistency is important, and duplicates should be _very_ rare. The standard is rather vague on this, but it's highly iffy for a server to duplicate `BrowseName`.